releasenotes/notes/add-elide-swaps-b0a4c373c9af1efd.yaml

```
@@ -0,0 +1,35 @@
+---
+features:
+  - |
+    Added a new optimization transpiler pass, :class:`~.ElideSwaps`,
+    which is designed to run prior to the :ref:`layout_stage` and will
+    optimize away any :class:`~.SwapGate`\s in a circuit by permuting virtual
+    qubits. For example, taking a circuit with :class:`~.SwapGate`\s:
+
+    .. plot::
+
+       from qiskit.circuit import QuantumCircuit
+
+       qc = QuantumCircuit(3)
+       qc.h(0)
+       qc.swap(0, 1)
+       qc.swap(2, 0)
+       qc.cx(1, 0)
+       qc.measure_all()
+       qc.draw("mpl")
+
+    will remove the swaps when the pass is run:
+
+    .. plot::
+
+       from qiskit.transpiler.passes import ElideSwaps
+       from qiskit.circuit import QuantumCircuit
+
+       qc = QuantumCircuit(3)
+       qc.h(0)
+       qc.swap(0, 1)
+       qc.swap(2, 0)
+       qc.cx(1, 0)
+       qc.measure_all()
+
+       ElideSwaps()(qc).draw("mpl")
```

In case you want to add this info:

```suggestion
       ElideSwaps()(qc).draw("mpl")
       
    This pass has been added to `optimization_level=3` in `generate_default_pass_manager()`.
```

---

This one isn't actually the case yet. We're doing this in #12111 , we split out from this PR to test it in isolation (ideally with more time before the release than we actually have now).